qualang_tools/plot/plot.py

```diff
@@ -1,17 +1,296 @@
+    """
+    Extract the simulated samples corresponding to a given element.
+
+    :param element_name: name of the element corresponding to the samples to extract. Must be defined in the config.
+    :param job: The simulated QmJob from which the samples will be extracted.
+    :param config: The config file used to create the job.
+    :return: 1D complex array containing the simulated samples for the specified element.
+    """
+    element = config["elements"][element_name]
+    sample_struct = job.get_simulated_samples()
+    if "mixInputs" in element:
+        port_i = element["mixInputs"]["I"]
+        port_q = element["mixInputs"]["Q"]
+        samples = (
+            sample_struct.__dict__[port_i[0]].analog[str(port_i[1])]
+            + 1j * sample_struct.__dict__[port_q[0]].analog[str(port_q[1])]
+        )
+    else:
+        port = element["singleInput"]["port"]
+        samples = sample_struct.__dict__[port[0]].analog[str(port[1])]
+    return samples
+
+
+def plot_simulator_output(
+    plot_axes: List[List[str]], job: QmJob, config: dict, duration_ns: int
+):
+    """
+    generate a 'plotly' plot of simulator output by elements
+
+    :param plot_axes: a list of lists of elements (ex: [["qubit0"], ["qubit1"]]). Will open
+    multiple axes, one for each list.
+    :param job: The simulated QmJob to plot.
+    :param config: The config file used to create the job.
+    :param duration_ns: the duration to plot in nanosecond.
+    :return: the generated 'plotly' figure.
+    """
+    time_vec = np.linspace(0, duration_ns - 1, duration_ns)
+    samples_struct = []
+    for plot_axis in plot_axes:
+        samples_struct.append(
+            [get_simulated_samples_by_element(pa, job, config) for pa in plot_axis]
+        )
+
+    fig = go.Figure().set_subplots(rows=len(plot_axes), cols=1, shared_xaxes=True)
+
+    for i, plot_axis in enumerate(plot_axes):
+        for j, plot_item in enumerate(plot_axis):
+            if samples_struct[i][j].dtype == np.float:
+                fig.add_trace(
+                    go.Scatter(x=time_vec, y=samples_struct[i][j], name=plot_item),
+                    row=i + 1,
+                    col=1,
+                )
+                print(samples_struct[i][j])
+            else:
+                fig.add_trace(
+                    go.Scatter(
+                        x=time_vec, y=samples_struct[i][j].real, name=plot_item + " I"
+                    ),
+                    row=i + 1,
+                    col=1,
+                )
+                fig.add_trace(
+                    go.Scatter(
+                        x=time_vec, y=samples_struct[i][j].imag, name=plot_item + " Q"
+                    ),
+                    row=i + 1,
+                    col=1,
+                )
+    fig.update_xaxes(title="time [ns]")
+    return fig
+
+
+def plot_ar_attempts(res_handles, qubits: List[int], attempts_data: str, **hist_kwargs):

```

Comment:
This is a very specific function, not sure what I think about it...
Does it matches our active reset code? Did we implement the repeat-until-success protocol in a manner which is compatible with this?
If so, we should link between the functions, this is the syntax
```
:func:`~qm.qua._dsl.declare`
```

Reply:
No we currently don't and I'll remove this function which was not asked by customers (it will remain in Lior's lib though)